gremlin/pom.xml

```diff
@@ -120,6 +126,24 @@
             <!--                    </execution>-->
             <!--                </executions>-->
             <!--            </plugin>-->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven-compiler-plugin.version}</version>
+		<configuration>
+                  <annotationProcessorPaths>
+                    <annotationProcessorPath>
+		      <groupId>org.apache.tinkerpop</groupId>
+		      <artifactId>gremlin-annotations</artifactId>
+		      <version>${gremlin.version}</version>
+                    </annotationProcessorPath>
+                  </annotationProcessorPaths>
+                  <annotationProcessors>
+                    <annotationProcessor>org.apache.tinkerpop.gremlin.process.traversal.dsl.GremlinDslProcessor</annotationProcessor>
```

Comment:
What does the annotation processor do? Does it browse all the jars in the classpath looking for an annotation?

Reply:
it builds the SocialTraversalSource from the SocialTraversalDsl for the tests:

https://tinkerpop.apache.org/docs/current/reference/#gremlin-java-dsl

Reply:
test only = good.